pandas/core/ops.py

```diff
@@ -745,35 +718,31 @@ def wrapper(left, right, name=name, na_op=na_op):
 
         converted = _Op.get_op(left, right, name, na_op)
 
-        left, right = converted.left, converted.right
         lvalues, rvalues = converted.lvalues, converted.rvalues
         dtype = converted.dtype
         wrap_results = converted.wrap_results
-        na_op = converted.na_op
 
         if isinstance(rvalues, ABCSeries):
-            name = _maybe_match_name(left, rvalues)
+            res_name = _maybe_match_name(left, rvalues)
             lvalues = getattr(lvalues, 'values', lvalues)
             rvalues = getattr(rvalues, 'values', rvalues)
             # _Op aligns left and right
         else:
             if isinstance(rvalues, pd.Index):
-                name = _maybe_match_name(left, rvalues)
+                res_name = _maybe_match_name(left, rvalues)
             else:
-                name = left.name
+                res_name = left.name
             if (hasattr(lvalues, 'values') and
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
-        result = wrap_results(safe_na_op(lvalues, rvalues))
-        return construct_result(
-            left,
-            result,
-            index=left.index,
-            name=name,
-            dtype=dtype,
-        )
+        result = wrap_results(safe_na_op(lvalues, rvalues, converted.na_op))
+        return construct_result(left, result,
+                                index=left.index, name=res_name, dtype=dtype)
 
+    wrapper.__name__ = name
+    if hasattr(operator, name) and callable(getattr(operator, name)):
```

Comment:
huh? what is this for

Reply:
You asked for generic docstrings for these ops.  These are about as generic as they get.  But there is no e.g. operator.rfloordiv, so we have to condition.

Reply:
I understand, but this is an obscure way of doing this. I would rather actually construct a proper doc-string here

Reply:
OK, I'll remove this, can consider docstrings out of scope for this PR.